### PR TITLE
adds list dataset handler and update/delete voices handler 

### PR DESCRIPTION
**Pull request by kasisnu**
I think I misunderstood what "private" meant for the list datasets handler but I'll revisit the logic so the visibility enum is respected.

Since right now the fields between the Dataset and Voice models are pretty similar the update/delete handlers are similar too. We can modify the fields as the feature evolves but I think the outlines are there.

### CODE REVIEW

File: crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+pub async fn list_datasets_with_connection(
+    mysql_connection: &mut PoolConnection<MySql>,
+    maybe_creator_token: Option<&str>,
+    can_see_deleted: bool,
+) -> AnyhowResult<Vec<DatasetRecordForList>> {
+
+    let maybe_models = match maybe_creator_token {
+        Some(username) => {
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs]: s/username/user_token

File: crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    pub creator_set_visibility: Visibility,
+    pub ietf_language_tag: String,
+    pub ietf_primary_language_subtag: String,
+    pub maybe_creator_user_token: Option<String>,
+
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+}
+
+pub async fn list_datasets_by_token(
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs]: maybe we should rename this to "list_datasets_by_user_token" and make it non-optional.

File: crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+) -> AnyhowResult<Vec<DatasetRecordForList>> {
+
+    let maybe_models = match maybe_creator_token {
+        Some(username) => {
+            list_datset_by_creator(mysql_connection, username, can_see_deleted)
+                .await
+        },
+        None => {
+            // TODO(kasisnu): anonymous users can list?
+            return Ok(Vec::new());
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/datasets/list_datasets.rs]: yeah, let's make this non-optional.

File: crates/lib/mysql_queries/src/queries/voice_designer/delete_voice.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        r#"
+UPDATE zs_voices
+SET
+  mod_deleted_at = CURRENT_TIMESTAMP,
+  maybe_mod_user_token = ?
+WHERE
+  token = ?
+LIMIT 1
+        "#,
+      mod_user_token,
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/delete_voice.rs]: excellent!

File: crates/lib/mysql_queries/src/queries/voice_designer/delete_voice.rs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      voice_token,
+    )
+        .execute(mysql_pool)
+        .await?;
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/delete_voice.rs]: this is all good!

File: crates/lib/mysql_queries/src/queries/voice_designer/get_voice.rs [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        zv.ietf_language_tag,
+        zv.ietf_primary_language_subtag,
+        zv.maybe_creator_user_token,
+        zv.creator_set_visibility as `creator_set_visibility: enums::common::visibility::Visibility`
+        FROM zs_voices as zv
+        WHERE
+            zv.token = ?
+            AND zv.user_deleted_at IS NULL
+            AND zv.mod_deleted_at IS NULL
+            "#,
```
Comment (echelon) [inline crates/lib/mysql_queries/src/queries/voice_designer/get_voice.rs]: perfect!

File: crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/delete_voice.rs [hunk 7]
```diff
@@ -1,13 +1,175 @@
...
+        undelete_voice_as_mod(
+          &path.voice_token,
+          &user_session.user_token,
+          &server_state.mysql_pool
+        ).await
+      }
+    }
+  };
+
+  match query_result {
```
Comment (echelon) [inline crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/delete_voice.rs:166]: Excellent! :D

File: crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/list_datasets_by_user.rs [hunk 8]
```diff
@@ -50,16 +59,72 @@ impl ResponseError for ListDatasetsByUserError {
...
+      warn!("not logged in");
+      return Err(ListDatasetsByUserError::NotAuthorized);
+    }
+  };
+
+  let user_token = path.user_token.clone();
+  let is_mod = user_session.can_ban_users;
+
+  // TODO(kasisnu): fix this so this matches "private" scoping - did that mean visibility cause the user token is in the path
+  let query_results = list_datasets_by_token(&server_state.mysql_pool, Some(&user_token), is_mod).await.map_err(|e| {
```
Comment (echelon) [inline crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/list_datasets_by_user.rs]: yeah, let's always assume user token is present

Comment (echelon) [reply crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/list_datasets_by_user.rs]: maybe what we want: 

- Users can see all of their own undeleted models
- Visitors to a profile can see all of another users' undeleted (and public) models
- Moderators can see everything

**Review by echelon — APPROVED** [review]
This is great! A few comments on list, but we can land as-is and follow up with changes.

File: crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/list_datasets_by_user.rs (hunk 8 above)
Comment (kasisnu) [reply crates/service/web/storyteller_web/src/http_server/endpoints/voice_designer/list_datasets_by_user.rs]: Ooh gotcha, ty for confirming! I made these changes _just_ now but they're a lil different(I'm filtering out the records in the map/collect chain) from the w2l/tts results query builder. Lmk if I should update the queries to match those builders

**Review by echelon — APPROVED** [review]

**Pull request merged**